File: 0031-next-permutation/0031-next-permutation.java
class Solution {
    public void nextPermutation(int[] nums) {
        int n = nums.length;
        boolean flag = true;

        for(int i = n - 2; i>=0; i--) {
            if(nums[i] < nums[i+1]) {
                flag = false;
                Arrays.sort(nums, i+1, n);
                int k = i+1;
                while(nums[i] >= nums[k]) k++;
                int temp = nums[i];
                nums[i] = nums[k];
                nums[k] = temp;
                break;
            }
        }

        if(flag) Arrays.sort(nums);
    }
}

File: 0031-next-permutation/0031-next-permutation.cpp
class Solution {
public:
    void nextPermutation(vector<int>& nums) {
        int n = nums.size();
        bool flag = true;

        for(int i = n - 2; i>=0; i--) {
            if(nums[i] < nums[i+1]) {
                flag = false;
                sort(nums.begin() + i + 1, nums.end());
                int k = i + 1;
                while(nums[i] >= nums[k]) k++;
                swap(nums[i], nums[k]);
                break;
            }
        }

        if(flag) sort(nums.begin(), nums.end());
    }
};